code/modules/client/preference_setup/general/02_language.dm

```diff
@@ -38,6 +38,10 @@
 	)
 
 /datum/category_item/player_setup_item/general/language/sanitize_character(var/sql_load = 0)
+
+	if(!pref.client)
```

Comment:
This is bad. Sanitize character is only meant to be invoked through player interaction. Ergo, a client is present. Doing this will introduce the possibility of an unsanitized character getting through, which is a security issue.

Can you post the full call stack of this runtime?

Reply:
```
runtime error:
[09:03:50]Cannot read null.mob
[09:03:50]proc name: sanitize character (/datum/category_item/player_setup_item/general/language/sanitize_character)
[09:03:50]  source file: 02_language.dm,59
[09:03:50]  usr: 0
[09:03:50]  src: Language (/datum/category_item/player_setup_item/general/language)
[09:03:50]  call stack:
[09:03:50]Language (/datum/category_item/player_setup_item/general/language): sanitize character(0)
[09:03:50]General (/datum/category_group/player_setup_category/general_preferences): sanitize setup(0)
[09:03:50]/datum/category_collection/pla... (/datum/category_collection/player_setup_collection): sanitize setup(0)
[09:03:50]/datum/preferences (/datum/preferences): copy to(Quentin King (/mob/living/carbon/human), 1)
[09:03:50]/datum/preferences (/datum/preferences): randomize appearance for(Quentin King (/mob/living/carbon/human))
[09:03:50]the bolt of change (/obj/item/projectile/change): wabbajack(Quentin King (/mob/living/carbon/human))
[09:03:50]the bolt of change (/obj/item/projectile/change): on hit(Quentin King (/mob/living/carbon/human), 0, "chest")
[09:03:50]Quentin King (/mob/living/carbon/human): bullet act(the bolt of change (/obj/item/projectile/change), "chest")
[09:03:50]Quentin King (/mob/living/carbon/human): bullet act(the bolt of change (/obj/item/projectile/change), "chest")
[09:03:50]the bolt of change (/obj/item/projectile/change): attack mob(Quentin King (/mob/living/carbon/human), 2, 0)
[09:03:50]the bolt of change (/obj/item/projectile/change): Collide(Quentin King (/mob/living/carbon/human))
[09:03:50]the floor (112,89,4) (/turf/simulated/floor/tiled/white): Enter(the bolt of change (/obj/item/projectile/change), the floor (112,88,4) (/turf/simulated/floor/tiled/white))
[09:03:50]the bolt of change (/obj/item/projectile/change): Move(the floor (112,89,4) (/turf/simulated/floor/tiled/white), 1)
[09:03:50]the bolt of change (/obj/item/projectile/change): Move(the floor (112,89,4) (/turf/simulated/floor/tiled/white), 1)
[09:03:50]the bolt of change (/obj/item/projectile/change): pixel move(1, 1, 0)
[09:03:50]the bolt of change (/obj/item/projectile/change): process()
[09:03:50]Projectiles (/datum/controller/subsystem/processing/projectiles): fire(0)
[09:03:50]Projectiles (/datum/controller/subsystem/processing/projectiles): ignite(0)
[09:03:50]Master (/datum/controller/master): RunQueue()
[09:03:50]Master (/datum/controller/master): Loop()
[09:03:50]Master (/datum/controller/master): StartProcessing(0)
```

Reply:
Yesss. The fix proposed is indeed invalid and is likely to introduce a security risk.

Since the use of `client` here is only for the sake of printing error messages, wrap those printouts with a `if (client)` clause.